dotnet/src/InternalUtilities/src/Diagnostics/Verify.cs

```diff
@@ -84,7 +84,7 @@ internal static void DirectoryExists(string path)
     /// Make sure every function parameter name is unique
     /// </summary>
     /// <param name="parameters">List of parameters</param>
-    internal static void ParametersUniqueness(IList<ParameterView> parameters)
+    internal static void ParametersUniqueness(IReadOnlyList<ParameterView> parameters)
```

Comment:
Just tried applying this suggestion, thanks!. Using a HashSet will enforce the distinctness of the args, but it will not enforce arg order (collection vs list), and it will silently drop duplicates - did it drop arg 2 or arg 4?

For the "View" structures, I think these should be as simple as possible, and we need not enforce any laws in these classes. Instead, they should be enforced in the classes that manage these registrations (upon registration) and throw on invalid data.